test/new-e2e/tests/otel/utils/pipelines_utils.go

```diff
@@ -0,0 +1,375 @@
+// Copyright 2016-present Datadog, Inc.
+
+// Package utils contains util functions for OTel e2e tests
+package utils
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	fakeintake "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+)
+
+const (
+	service              = "telemetrygen-job"
+	env                  = "e2e"
+	version              = "1.0"
+	customAttribute      = "custom.attribute"
+	customAttributeValue = "true"
+)
+
+// OTelTestSuite is an interface for the OTel e2e test suite.
+type OTelTestSuite interface {
+	T() *testing.T
+	Env() *environments.Kubernetes
+}
+
+// TestTraces tests that OTLP traces are received through OTel pipelines as expected
+func TestTraces(s OTelTestSuite) {
+	ctx := context.Background()
+	err := s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+	require.NoError(s.T(), err)
+	numTraces := 10
+
+	s.T().Log("Starting telemetrygen")
+	createTelemetrygenJob(ctx, s, "traces", []string{"--traces", fmt.Sprint(numTraces)})
+
+	var traces []*aggregator.TracePayload
+	s.T().Log("Waiting for traces")
+	require.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		traces, err = s.Env().FakeIntake.Client().GetTraces()
+		assert.NoError(c, err)
+		assert.NotEmpty(c, traces)
+	}, 2*time.Minute, 10*time.Second)
+	require.NotEmpty(s.T(), traces)
+	s.T().Log("Got traces", traces)
+	s.T().Log("num traces", len(traces))
+	trace := traces[0]
+	require.NotEmpty(s.T(), trace.TracerPayloads)
+
+	// Verify tags on traces and spans
+	tp := trace.TracerPayloads[0]
+	ctags, ok := getContainerTags(s.T(), tp)
+	require.True(s.T(), ok)
+	assert.Equal(s.T(), env, tp.Env)
+	assert.Equal(s.T(), version, tp.AppVersion)
+	require.NotEmpty(s.T(), tp.Chunks)
+	require.NotEmpty(s.T(), tp.Chunks[0].Spans)
+	spans := tp.Chunks[0].Spans
+	for _, sp := range spans {
+		assert.Equal(s.T(), service, sp.Service)
+		assert.Equal(s.T(), env, sp.Meta["env"])
+		assert.Equal(s.T(), version, sp.Meta["version"])
+		assert.Equal(s.T(), customAttributeValue, sp.Meta[customAttribute])
+		assert.Equal(s.T(), "telemetrygen", sp.Meta["otel.library.name"])
+		assert.Equal(s.T(), sp.Meta["k8s.node.name"], tp.Hostname)
+
+		// Verify container tags from infraattributes processor
+		assert.NotNil(s.T(), ctags["kube_container_name"])
+		assert.NotNil(s.T(), ctags["kube_namespace"])
+		assert.NotNil(s.T(), ctags["pod_name"])
+		assert.Equal(s.T(), sp.Meta["k8s.container.name"], ctags["kube_container_name"])
+		assert.Equal(s.T(), sp.Meta["k8s.namespace.name"], ctags["kube_namespace"])
+		assert.Equal(s.T(), sp.Meta["k8s.pod.name"], ctags["pod_name"])
```

Comment:
We set these on SDK, can we test specific ones that infra attributes processor adds ?

Reply:
These are all the tags added by the infra attributes processor, if we want the infra attributes processor to add more tags we need to set more Kubernetes attributes on the SDK
```
tags:"custom.attribute:true" tags:"deployment.environment:e2e" tags:"host.name:kind-control-plane" tags:"k8s.container.name:telemetrygen-job" tags:"k8s.namespace.name:datadog" tags:"k8s.node.name:kind-control-plane" tags:"k8s.pod.name:telemetrygen-job-metrics-fv6tn" tags:"service.name:telemetrygen-job" tags:"service.version:1.0" tags:"env:e2e" tags:"service:telemetrygen-job" tags:"version:1.0" tags:"kube_container_name:telemetrygen-job" tags:"kube_namespace:datadog" tags:"pod_name:telemetrygen-job-metrics-fv6tn"
```